src/pykoala/corrections/sky.py

```diff
@@ -71,443 +187,640 @@ def pol_continuum(data, wavelength, pol_order=3, **polfit_kwargs):
+        variance : np.ndarray
+            Array of variance data to include errors in determining the sky.
+        axis : int, optional
+            Spectral direction of the data. Default is -1.
+
         Returns
         -------
-        data_subs: (np.ndarray)
-        var_subs: (np.ndarray)
+        data_subs : np.ndarray
+            Data array after the sky model has been subtracted.
+        var_subs : np.ndarray
+            Variance array after including the sky model variance.
         """
-        # TODO
         if data.ndim == 3 and self.intensity.ndim == 1:
             skymodel_intensity = self.intensity[:, np.newaxis, np.newaxis]
             skymodel_var = self.intensity[:, np.newaxis, np.newaxis]
         elif data.ndim == 2 and self.intensity.ndim == 1:
             skymodel_intensity = self.intensity[np.newaxis, :]
             skymodel_var = self.intensity[np.newaxis, :]
+        elif data.ndim == 2 and self.intensity.ndim == 2:  
+            skymodel_intensity = self.intensity       
+            skymodel_var = self.variance
         else:
-            self.vprint(f"Data dimensions ({data.shape}) cannot be reconciled with sky mode ({self.intensity.shape})")
+            self.vprint(
+                f"Data dimensions ({data.shape}) cannot be reconciled with "
+                + f"sky mode ({self.intensity.shape})")
         data_subs = data - skymodel_intensity
         var_subs = variance + skymodel_var
         return data_subs, var_subs
 
     def substract_pca():
-        # TODO: Implement PCA substraction method
+        """
+        Placeholder for PCA subtraction method.
+
+        This method is not yet implemented.
+        """
         pass
     
     def vprint(self, *messages):
-        """Print a message"""
+        """
+        Print messages if `verbose` is True.
+
+        Parameters
+        ----------
+        *messages : str
+            Messages to be printed.
+        """
         if self.verbose:
             print("[SkyModel] ", *messages)
 
 
 
 class SkyOffset(SkyModel):
     """
-    Sky model based on a single RSS offset sky exposure
+    Sky model based on a single RSS offset sky exposure.
 
-    Description
-    -----------
     This class builds a sky emission model from individual sky exposures.
 
     Attributes
     ----------
-    - dc:
-        Data container used to estimate the sky
-    - exptime:
-        Data container net exposure time
+    dc : DataContainer
+        Data container used to estimate the sky.
+    exptime : float
+        Net exposure time from the data container.
     """
+
     def __init__(self, dc):
         """
+        Initialize the SkyOffset model with a data container.
 
         Parameters
         ----------
-        rss: RSS
-            Raw Stacked Spectra corresponding to the offset-sky exposure.
+        dc : DataContainer
+            Raw Stacked Spectra (RSS) corresponding to the offset-sky exposure.
         """
```

Comment:
Row- (not Raw- ;^D) Stacked Spectra (`RSS`) or `DataContainer`? I don't think anyone should ever estimate the sky from a `Cube`, unless it was (for some weird reason) the only option allowed... and, in that case, I wouldn't mind forcing the user to create an `RSS`, to make it clear ;^D

Reply:
Why not? If the sky is stable between the RSS exposures it would not make a big difference, right? Besides, the cube could come from a single RSS.